src/Components/Web.JS/src/Rendering/BrowserRenderer.ts

```diff
@@ -377,6 +381,14 @@ export function isInteractiveRootComponentElement(element: LogicalElement): bool
   return element[interactiveRootComponentPropname];
 }
 
+export function setShouldPreserveContentOnInteractiveComponentDisposal(element: LogicalElement, shouldPreserve: boolean) {
+  element[preserveContentOnDisposalPropname] = shouldPreserve;
+}
+
+function shouldPreserveContentOnInteractiveComponentDisposal(element: LogicalElement): boolean {
+  return element[preserveContentOnDisposalPropname] === true;
+}
+
```

Comment:
If this mechanism is as simple as it looks from my naive reading of it, this is a pretty neat technique.